pubsub/google/cloud/pubsub_v1/subscriber/policy/thread.py

```diff
@@ -332,8 +318,7 @@ def on_response(self, response):
         """
         for msg in response.received_messages:
             _LOGGER.debug(
-                'Using %s to process new message received:\n%r',
-                self._callback, msg)
+                'Using %s to process message with ack_id %s.',
+                self._callback, msg.ack_id)
             message = Message(msg.message, msg.ack_id, self._request_queue)
-            future = self._executor.submit(self._callback, message)
```

Comment:
Can you explain what this change fixes? It looks suspect for unintended consequences.

Reply:
This actually causes unnecessary memory and CPU to be consumed just for a log output, especially with a large amount of pending messages..

Reply:
Ack.